### PR TITLE
Obg 736: payment integration tests

### PR DESCRIPTION
**Pull request by ivaniuminov**
https://github.com/adorsys/open-banking-gateway/issues/736

### CODE REVIEW

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java [hunk 1]
```diff
@@ -75,6 +75,8 @@ private String calculateSignature(RequestTemplate requestTemplate, Instant insta
                 return requestSigningService.signature(mapper.mapToBankProfile(headers, instant));
             case CONFIRM_CONSENT:
                 return requestSigningService.signature(mapper.mapToConfirmConsent(headers, instant));
+            case PIS:
+                return requestSigningService.signature(mapper.mapToPaymenyInitiation(headers, instant));
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java:79]: typo 'mapToPaymenyInitiation' -> mapToPaymentInitiation

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/RsaJwtsVerifyingServiceImpl.java [hunk 2]
```diff
@@ -84,6 +82,24 @@ public boolean verify(String signature, String encodedPublicKey, ConfirmConsentD
         ));
     }
 
+    @Override
+    public boolean verify(String signature, String encodedPublicKey, PaymentInitiationDataToSign data) {
+        return createDataToSign(signature, encodedPublicKey, data.getBankId(), data.getFintechUserId(), data.getRedirectOk(), data.getRedirectNok(), data.getXRequestId(), data.getInstant(),
+                                data.getOperationType());
+    }
+
+    private boolean createDataToSign(String signature, String encodedPublicKey, String bankId, String fintechUserId, String redirectOk, String redirectNok, UUID xRequestId, Instant instant,
```
Comment (valb3r) [inline opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/RsaJwtsVerifyingServiceImpl.java:91]: Looks like body is mostly duplicate to
https://github.com/adorsys/open-banking-gateway/pull/738/files#diff-a4d480f039ef3160a509a35345f1ce31R103

maybe we could extract:
```
Map<String, String> values = new HashMap<>();
        values.put(HttpHeaders.BANK_ID, bankId);
        values.put(HttpHeaders.FINTECH_USER_ID, fintechUserId);
        values.put(HttpHeaders.FINTECH_REDIRECT_URL_OK, redirectOk);
        values.put(HttpHeaders.FINTECH_REDIRECT_URL_NOK, redirectNok);
        DataToSign dataToSign = new DataToSign(xRequestId, instant, operationType, values);
```
to have DataToSignFactory or something similar

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/external/mapper/HttpRequestToDataToSignMapper.java [hunk 3]
```diff
@@ -75,4 +76,19 @@ public ConfirmConsentDataToSign mapToConfirmConsent(HttpServletRequest request,
 
         return new ConfirmConsentDataToSign(UUID.fromString(xRequestId), instant, OperationType.valueOf(operationType));
     }
+
+    public PaymentInitiationDataToSign mapToPaymentInititation(HttpServletRequest request, Instant instant) {
+        String xRequestId = request.getHeader(HttpHeaders.X_REQUEST_ID);
+        String operationType = request.getHeader(HttpHeaders.X_OPERATION_TYPE);
+
+        return new PaymentInitiationDataToSign(
```
Comment (valb3r) [inline opba-api-security/src/main/java/de/adorsys/opba/api/security/external/mapper/HttpRequestToDataToSignMapper.java:84]: Looks like payment amount and currency is missing from signature - if it is the case  then we should create a bug

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentInitiationRequestCommon.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    protected String serviceSessionId;
+
+    @ProvidedScenarioState
+    protected String redirectCode;
+
+    @Autowired
+    private RequestSigningService requestSigningService;
+
+    public SELF fintech_calls_initiate_payment_for_anton_brueckner() {
+        ExtractableResponse<Response> response = withPaymentHeaders(ANTON_BRUECKNER, requestSigningService, OperationType.PIS)
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentInitiationRequestCommon.java:41]: Please follow
```
given()
   .action()
.when()
   .action()
```
formatting

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PisStagesCommonUtil.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                       .header(FINTECH_USER_ID, fintechUserId)
+                       .header(FINTECH_ID, DEFAULT_FINTECH_ID)
+                       .header(X_REQUEST_ID, xRequestId.toString())
+                       .header(X_TIMESTAMP_UTC, xTimestampUtc.toString())
+                       .header(X_OPERATION_TYPE, operationType);
+    }
+
+    private static String calculatePaymentSignature(RequestSigningService requestSigningService, UUID xRequestId, Instant xTimestampUtc,
+                                                    OperationType operationType, String fintechUserId) {
+        PaymentInitiationDataToSign aymentInitiationDataToSign = new PaymentInitiationDataToSign(xRequestId, xTimestampUtc, operationType,
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PisStagesCommonUtil.java:72]: typo `aymentInitiationDataToSign` -> paymentInitiationDataToSign

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java (hunk 1 above)
Comment (ivaniuminov) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java]: renamed, thanks

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/external/mapper/HttpRequestToDataToSignMapper.java (hunk 3 above)
Comment (ivaniuminov) [reply opba-api-security/src/main/java/de/adorsys/opba/api/security/external/mapper/HttpRequestToDataToSignMapper.java:84]: Issue was created. https://github.com/adorsys/open-banking-gateway/issues/747

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentInitiationRequestCommon.java (hunk 4 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentInitiationRequestCommon.java]: This part of code was formatted

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PisStagesCommonUtil.java (hunk 5 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PisStagesCommonUtil.java]: Renamed, thanks

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentStagesCommonUtil.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    public static final String ANTON_BRUECKNER = "anton.brueckner";
+    public static final String SESSION_PASSWORD = "qwerty";
+    public static final String COMPUTE_IP_ADDRESS = "false";
+    public static final String IP_ADDRESS = "1.1.1.1";
+
+    public static RequestSpecification withPaymentHeaders(String fintechUserId, RequestSigningService requestSigningService, OperationType operationType) {
+        UUID xRequestId = UUID.randomUUID();
+        Instant xTimestampUtc = Instant.now();
+
+        return RestAssured
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentStagesCommonUtil.java:53]: Please follow:
```
given()
   .header()
```
formatting

Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentStagesCommonUtil.java:54]: Done

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/RsaJwtsVerifyingServiceImpl.java (hunk 2 above)
Comment (ivaniuminov) [reply opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/RsaJwtsVerifyingServiceImpl.java:91]: DataToSignFactory  will be implemented as a part of https://github.com/adorsys/open-banking-gateway/pull/773 (the reason is that this pull request is about tests,  and there will be implemented complete data signing logic)

**Pull request merged**